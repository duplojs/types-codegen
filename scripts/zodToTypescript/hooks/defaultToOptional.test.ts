import { ZodToTypescript } from "@duplojs/zod-to-typescript";
import { defaultToOptionalHook } from "./defaultToOptional";
import { zod } from "@duplojs/core";

it("defaultToOptionalHook", () => {
	const zodToTypescript = new ZodToTypescript();

	zodToTypescript.zodSchemaHooks.push(
		defaultToOptionalHook,
	);

	zodToTypescript.append(
		zod.string().default("toto"),
		"test",
	);

	expect(zodToTypescript.toString())
		.toMatchSnapshot();
});
