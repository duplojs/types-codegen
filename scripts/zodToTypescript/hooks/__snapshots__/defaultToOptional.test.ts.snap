// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaultToOptionalHook 1`] = `"type test = string | undefined;"`;
